internal/request/graphql/schema/generate.go

```diff
@@ -1327,6 +1392,104 @@ func (g *Generator) genLeafFilterArgInput(obj gql.Type) *gql.InputObject {
 	return selfRefType
 }
 
+// genEncryptedFilterArgInput generates filter input that only supports _eq on encrypted fields
+func (g *Generator) genEncryptedFilterArgInput(
+	obj *gql.Object,
+	encryptedIndexes []client.EncryptedIndexDescription,
+) *gql.InputObject {
+	inputCfg := gql.InputObjectConfig{
+		Name: g.genEncryptedFilterTypeName(obj),
+	}
+
+	fieldThunk := (gql.InputObjectConfigFieldMapThunk)(
+		func() (gql.InputObjectConfigFieldMap, error) {
+			fields := gql.InputObjectConfigFieldMap{}
+
+			for _, encIdx := range encryptedIndexes {
+				var objField *gql.FieldDefinition
+				for f, field := range obj.Fields() {
+					if f == encIdx.FieldName {
+						objField = field
+						break
+					}
+				}
+				if objField == nil {
+					// This should not happen, as we should have already validated the schema
```

Comment:
todo: I would much prefer a panic to pretending everything is fine here, it should make fixing whatever bug led to this being nil slightly easier.  As well reducing the risk to the users system where they may be saving a corrupt state.